flink-rpc/flink-rpc-akka-loader/src/main/java/org/apache/flink/runtime/rpc/akka/AkkaRpcSystemLoader.java

```diff
@@ -55,7 +55,13 @@ public RpcSystem loadRpcSystem(Configuration config) {
             final ClassLoader flinkClassLoader = RpcSystem.class.getClassLoader();
 
             final Path tmpDirectory = Paths.get(ConfigurationUtils.parseTempDirectories(config)[0]);
-            Files.createDirectories(tmpDirectory);
+            if (!Files.isSymbolicLink(tmpDirectory)) {
+                // Create tmp dir if it doesn't exist.
+                Files.createDirectories(tmpDirectory);
+            } else if (!Files.exists(tmpDirectory)) {
+                // For symbolic link, create the linked dir if it doesn't exist.
+                Files.createDirectory(Files.readSymbolicLink(tmpDirectory));
+            }
```

Comment:
Logic like this doesn't belong in here, but in some re-usable file util.

Reply:
Make sense, i will move this logical to a file util.